public/app/features/dashboard-scene/panel-edit/VizPanelManager.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+
+import { FieldConfigSource, PanelModel } from '@grafana/data';
+import {
+  SceneObjectState,
+  VizPanel,
+  SceneObjectBase,
+  SceneComponentProps,
+  sceneUtils,
+  DeepPartial,
+} from '@grafana/scenes';
+
+interface VizPanelManagerState extends SceneObjectState {
+  panel: VizPanel;
+}
+
+export class VizPanelManager extends SceneObjectBase<VizPanelManagerState> {
+  private _cachedPluginOptions: Record<
+    string,
+    { options: DeepPartial<{}>; fieldConfig: FieldConfigSource<DeepPartial<{}>> } | undefined
+  > = {};
+
+  public constructor(panel: VizPanel) {
+    super({ panel });
+  }
+
+  public changePluginType(pluginType: string) {
+    const {
+      options: prevOptions,
+      fieldConfig: prevFieldConfig,
+      pluginId: prevPluginId,
+      ...restOfOldState
+    } = sceneUtils.cloneSceneObjectState(this.state.panel.state);
+    this._cachedPluginOptions[prevPluginId] = { options: prevOptions, fieldConfig: prevFieldConfig };
+    const cachedOptions = this._cachedPluginOptions[pluginType]?.options;
+    const cachedFieldConfig = this._cachedPluginOptions[pluginType]?.fieldConfig;
+
+    const newPanel = new VizPanel({
+      options: { ...prevOptions, ...cachedOptions },
+      fieldConfig: { ...prevFieldConfig, ...cachedFieldConfig },
+      ...restOfOldState,
+      pluginId: pluginType,
+    });
+
+    const panel: PanelModel = {
+      title: newPanel.state.title,
+      options: newPanel.state.options,
+      fieldConfig: newPanel.state.fieldConfig,
+      id: 1,
+      type: pluginType,
+    };
+    const newOptions = newPanel.getPlugin()?.onPanelTypeChanged?.(panel, prevPluginId, prevOptions, prevFieldConfig);
+    if (newOptions) {
+      newPanel.onOptionsChange(newOptions, true);
+    }
+
+    this.setState({ panel: newPanel });
+  }
+
+  public static Component = ({ model }: SceneComponentProps<VizPanelManager>) => {
+    const { panel } = model.useState();
+
+    return <panel.Component model={panel} />;
```

Comment:
So this will cause queries to be re-run when cloned panel is rendered - this is because in the render path of the viz panel we are [setting SQR's container width](https://github.com/grafana/scenes/blob/main/packages/scenes/src/querying/SceneQueryRunner.ts#L284) that's triggering the queries to be performed again unnecessarily. 

Container width in SQR is an instance property, not state property, so it's not clonable. At the same time when chaning the type the container width remain the same, so I think we gonna need to extend interface of `setContainerWidth` to allow skipping query execution. 

Or alternatively - we need to revisit https://github.com/grafana/scenes/pull/334 by @torkelo, pause auto queries execution in edit mode and manually trigger queries when needed (i.e. i.e. queries/transformations  edit). Think that would be the desired path forward to address this problem. Thoughts @torkelo ?

Reply:
@dprokop hm.. what a coincidence, had the same problem with the new view panel mode I am working on.

Solved it with this: https://github.com/grafana/scenes/pull/482